ros_control:
  # specifies which information should be read
  read_position: true
  read_velocity: true
  read_effort: true
  read_volt_temp: true # this also corresponds for the error byte
  VT_update_rate: 100 # how many normal (position) reads have to be performed before one time the temperature, voltage and error is read
  warn_temp: 60
  read_imu: true
  control_loop_hz: 200

  # specification of the connected dynamixel servos
  dynamixels:
    port_info:
      port_name: /dev/ttyACM0
      baudrate: 2000000
      protocol_version: 2
    control_mode: position
    auto_torque: true
    device_info:
      RShoulderPitch:
        id: 1
        model_number: 311
        mounting_offset: 0.0
        offset: 0.0
      LShoulderPitch:
        id: 2
        model_number: 311
      RShoulderRoll:
        id: 3
        model_number: 321
      LShoulderRoll:
        id: 4
        model_number: 311
      RElbow:
        id: 5
        model_number: 321
      LElbow:
        id: 6
        model_number: 311
      RHipYaw:
        id: 7
        model_number: 321
      LHipYaw:
        id: 8
        model_number: 321
      RHipRoll:
        id: 9
        model_number: 321
      LHipRoll:
        id: 10
        model_number: 321
      RHipPitch:
        id: 11
        model_number: 321
      LHipPitch:
        id: 12
        model_number: 321
      RKnee:
        id: 13
        model_number: 321
      LKnee:
        id: 14
        model_number: 321
      RAnklePitch:
        id: 15
        model_number: 321
      LAnklePitch:
        id: 16
        model_number: 321
      RAnkleRoll:
        id: 17
        model_number: 321
      LAnkleRoll:
        id: 18
        model_number: 321
      HeadPan:
        id: 19
        model_number: 311
      HeadTilt:
        id: 20
        model_number: 311
      

  IMU:
    name: imu/data_raw
    frame: torso
